Configure.cmake

```diff
@@ -42,6 +45,28 @@ CHECK_C_SOURCE_COMPILES("
     __m128d r = _mm_mul_pd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
   COMPILER_SUPPORTS_SSE2)
 
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_SSE4})
+CHECK_C_SOURCE_COMPILES("
+  #if defined(_MSC_VER)
+  #include <intrin.h>
+  #else
+  #include <x86intrin.h>
+  #endif
+  int main() {
+    __m128d r = _mm_floor_sd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
+  COMPILER_SUPPORTS_SSE4)
+
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_AVX})
```

Comment:
AVX is not available for Cygwin because its ABI does not have support yet. I think it is okay to add detection for that here. It does not make much sense to do detection using CHECK_C_SOURCE_COMPILES here, since all the supported compiler should compile this.

One way to detect if there is ABI support is to check if the compiler is Cygwin.
Another way is to check if the compiled program runs correctly.
The following program runs normally on linux, but segfaults on Cygwin.

#include <stdio.h>
#include <x86intrin.h>

__m256d check() {
  return _mm256_set1_pd(1);
}

int main() {
  __m256d v = check();
  double a[4];
  _mm256_storeu_pd(a, v);
  printf("%g %g %g %g\n", a[0], a[1], a[2], a[3]);
}

If we think about cross-compiling, detecting Cygwin compiler would be better.
If we think about future possibility for this problem to be fixed, checking if the program runs would be better.